website/templates/profile/oauth_app_detail.mako

```diff
@@ -0,0 +1,93 @@
+                <li><a href="${ web_url_for('user_account') }">Account Settings</a></li>
+                <li><a href="${ web_url_for('user_addons') }">Configure Add-ons</a></li>
+                <li><a href="${ web_url_for('user_notifications') }">Notifications</a></li>
+                <li><a href="${ web_url_for('oauth_application_config') }">Developer apps</a></li>
+            </ul>
+        </div><!-- end sidebar -->
+    </div>
+
+    <div class="col-sm-9 col-md-7">
+
+        <div id="app-detail">
+            <div id="app-keys" class="border-box text-right text-muted"
+                 data-bind="visible: dataUrl">
+                <p><span><strong>Client ID</strong>:</span> <br><span data-bind="text: content().clientId"></span></p>
+                <p><span><strong>Client secret</strong>:</span> <br><span data-bind="text: content().clientSecret"></span></p>
+            </div>
+            <div id="app-fields">
+                <!-- TODO: Add revoke/ reset buttons -->
+                <form role="form">
+                    <div class="form-group">
+                        <label>Application name</label>
+                        <input class="form-control" type="text" data-bind="value: content().name">
+                    </div>
+
+                    <div class="form-group">
+                        <label>Project homepage URL</label>
+                        <input class="form-control" type="text" data-bind="value: content().homeUrl">
+                    </div>
+
+                    <div class="form-group">
+                        <label>Application description</label>
+                        <textarea class="form-control" placeholder="Application description is optional" data-bind="value: content().description"></textarea>
+                    </div>
+
+                    <div class="form-group">
+                        <label>Authorization callback URL</label>
+                        <input type="text" class="form-control" data-bind="value: content().callbackUrl">
+                    </div>
+
+                    <div class="padded">
+                        <button type="submit" class="btn btn-primary"
+                                data-bind="visible: !dataUrl, click: $root.createApplication">Create</button>
+
+                        <button type="submit" class="btn btn-primary"
+                                data-bind="visible: dataUrl, click: $root.updateApplication">Update</button>
+                    </div>
+                </form>
+
+                <!-- Flashed Messages -->
+                <div class="help-block">
+                    <p data-bind="html: message, attr.class: messageClass"></p>
+                </div>
+            </div>
+        </div> <!-- End app-detail section -->
+    </div>
+</div>
+</%def>
+
+<%def name="javascript_bottom()">
+<script type="text/javascript">
+   ## Store mako variables on window so they are accessible from JS
+   ## modules. Not sure if this is a good idea.
```

Comment:
Ha, I think my comments were copy-and-pasted...

But yeah, still not sure if it's a good idea. =)